ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -210,6 +212,13 @@ public void onStop(final Consumer<Boolean> onStop) {
     return StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.REPLACE_THREAD;
   }
 
+  public Set<TaskMetadata> getTaskMetadata() {
+    return kafkaStreams.localThreadsMetadata()
+                       .stream()
+                       .flatMap(t -> t.activeTasks().stream())
+                       .collect(Collectors.toSet());
```

Comment:
Shouldn't it be better if an aggregation is returned here? Or are there plans to look at each task of the stream per query? 

Say a user has 100 queries running, how many tasks are expected per query (in average)? Would this make the healthcheck slow if the list is high?

Reply:
There are plans to look at each task. It really depends on how complicated the query is but there should not be too many tasks and the taskMetadata is not too heavy of an object.